sdk/identity/identity/src/credentials/deviceCodeCredential.ts

```diff
@@ -120,8 +127,11 @@ export class DeviceCodeCredential implements TokenCredential {
       options,
       async (newOptions) => {
         const arrayScopes = Array.isArray(scopes) ? scopes : [scopes];
-        await this.msalFlow.getToken(arrayScopes, newOptions);
-        return this.msalFlow.getActiveAccount();
+        await this.msalClient.getTokenByDeviceCode(arrayScopes, this.userPromptCallback, {
+          ...newOptions,
+          disableAutomaticAuthentication: false, // this method should always allow user interaction
```

Comment:
I understand that we want to not attempt silent authentication, but the naming of this parameter confuses me since you'd think you'd *want* to disable automatic authentication rather than setting this to false which would enable it?

Reply:
Yeah, I agree that the naming is confusing. The reason I stuck with it is because it matches the same behavior and meaning as the public API's version of [`disableAutomaticAuthentication`](https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/identity/identity/review/identity.api.md?plain=1#L348) which can help when searching the codebase for this.

Personally I think a name like `disableInteractiveAuthentication` makes more sense, but I'm not sure if I want to stay consistent with the meaning of the public API field or do the name-mapping here

What this actually does, if I'm reading the [source code](https://github.com/Azure/azure-sdk-for-js/blob/455f2d5495bcfa71b5af2e1cfd0f80c848af1e16/sdk/identity/identity/src/msal/nodeFlows/msalNodeCommon.ts#L456-L468) correctly is disable interactive auth - if an AuthenticationRequiredError is thrown from getTokenSilent and this option is true we will throw instead of continuing on to the interactive auth. 

With that context in place - I don't have strong feelings and am happy to change it, but I have a _slight_ preference to keeping the name and meaning consistent with the public API for better or worse. What do you think? @KarishmaGhiya would love your take as well!

Reply:
@maorleger  Is this param on L132 actually set by the public API param (L105) ? Or is this used for disabling interactive auth in another scenario (other than when passed in as a parameter by the user) ?

Reply:
This is used for _enabling_ interactive auth fallback when the user passes is `disableAutomaticAuthentication: true` - so in the `getToken` call we will simply pass whatever the user passed in the public API param. In `authenticate` we ignore what the user passed in because `authenticate` is meant to be called by the user when `getToken` fails to silent auth, _and_ disableAutomaticAuthentication is true.

Here's a few scenarios:


| user sets disableAutomaticAuthentication | deviceCodeCredential#getToken | deviceCodeCredential#authenticate |
| ---- | ---- | ---- |
| undefined | false | false | 
| false | false | false |
| true | true | false | 

Happy to discuss more in our sync!